### PR TITLE
[Improvement] Refactor all printed NeuroKit warnings into warning control

### PR DESCRIPTION
**Pull request by awwong1**
# Description

This PR shows how to migrate the `print` calls to `warnings.warn`.
- https://github.com/neuropsychology/NeuroKit/issues/316
- https://docs.python.org/3/library/warnings.html#warnings.warn
- https://docs.pytest.org/en/latest/warnings.html#recwarn

# Proposed Changes

I replaced one warning in `signal_psd`, @DominiqueMakowski
> open a PR with like maybe one example and then we can see how it looks like, and if it works we go ahead to improve it throughout

Here's how it'll work if you call the function:

```python
import neurokit2 as nk

data = nk.data("bio_eventrelated_100hz")

out = nk.signal_psd(data["ECG"], sampling_rate=100)
```
```text
/home/alexander/sandbox/src/github.com/awwong1/NeuroKit/neurokit2/signal/signal_psd.py:112: NeuroKitWarning: signal_psd(): The duration of recording is too short to support a sufficiently long window for high frequency resolution. Consider using a longer recording or increasing the `min_frequency`
  category=NeuroKitWarning
```

# Checklist

Here are some things to check before creating the PR.

- [X] I have read the [CONTRIBUTING](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#structure-and-code) file.
- [X] My PR is targetted at the **dev branch** (and not towards the master branch).
- [X] I ran the [CODE CHECKS](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#run-code-checks) on the files I added or modified and fixed the errors.

### CODE REVIEW

**Comment by DominiqueMakowski** [comment]
Looks good, I'd say we schedule this improvement for the next release (as we'll probably merge the current one today or asap).

From that example you mentioned it seems indeed as repeating the function is not necessarily as it shows the line and the file... what do you think?

Also, is it possible to wrap this:
```python
warn(
                "signal_psd(): The duration of recording is too short to support a "
                "sufficiently long window for high frequency resolution. Consider using a longer recording "
                "or increasing the `min_frequency`"
                "or increasing the `min_frequency`",
                category=NeuroKitWarning
            )
```

in a function like `throw_warning(message)`? I'm not sure about the benefits other than having not to write the category arg and also having not to import everytime the warnings module (tho we would still need to import the warnings where `throw_warning` would be defined, so it's not a big gain)

**Comment by awwong1** [comment]
The issue of wrapping the `warnings.warn` call in another function is the warning line number that appears will not reflect the location of the actual warning.

e.g. if `misc/warnings.py` had:
```python3
# -*- coding: utf-8 -*-
from warnings import warn

class NeuroKitWarning(RuntimeWarning):
    """Category for runtime warnings that occur within the NeuroKit library.
    """


def throw_warning(message):
    warn(message, category=NeuroKitWarning)
```

calling `throw_warning('whatever')` will show the error coming from `...neurokit2/misc/warnings.py:10:` no matter where you call the wrapper function.

Would you like me to change the rest of the print warnings? I will not modify any other neurokit warnings (such as the ImportErrors).

**Comment by DominiqueMakowski** [comment]
> The issue of wrapping the warnings.warn call in another function is the warning line number that appears will not reflect the location of the actual warning.

Ah right yes! in that case indeed doesn't make sense to wrap it indeed.

> Would you like me to change the rest of the print warnings? I will not modify any other neurokit warnings (such as the ImportErrors).

Yes I think we can go ahead and do that, with removing the function name from the warning message (as it's redundant). Thanks!

**Comment by DominiqueMakowski** [comment]
@awwong1 do you want me to merge this PR so we can open a new one with more or keep pushing to this one?

**Comment by awwong1** [comment]
Let's keep this PR open, I plan on having the changes finished by the end of day.

**Comment by awwong1** [comment]
> @awwong1 Thanks so much for your improvements, especially the incorporation of pytest warnings in our test files! Just picked up on these two errors and once these are changed we can merge smile

I am grateful that you took the time to review my changes. Thank you for humoring my print statement nitpicking. :smile_cat:

File: neurokit2/eog/eog_plot.py [hunk 1]
```diff
@@ -49,7 +49,10 @@ def eog_plot(eog_signals, peaks=None, sampling_rate=None):
 
     # Sanity-check input.
     if not isinstance(eog_signals, pd.DataFrame):
-        print("NeuroKit error: The `eog_signals` argument must be the DataFrame returned by `eog_process()`.")
+        raise ValueError(
```
Comment (zen-juen) [inline neurokit2/eog/eog_plot.py:52]: can add '*eog_plot()*' in "NeuroKit error: eog_plot(): The `eog_signals` ...."

Comment (awwong1) [reply neurokit2/eog/eog_plot.py:52]: Good catch, will modify + add test

**Pull request merged**